lib_device_control/host/device_access_usb.c

```diff
@@ -175,6 +176,64 @@ control_read_command(control_resid_t resid, control_cmd_t cmd,
   return CONTROL_SUCCESS;
 }
 
+control_ret_t list_connected_devices(int vendor_id)
+{
+  uint32_t pid_list[MAX_NUM_DEVICES];
+  uint8_t num_devices = 0;
+  uint32_t current_pid = 0;
+  #ifdef _WIN32
+  usb_init();
+  usb_find_busses(); /* find all busses */
+  usb_find_devices(); /* find all connected devices */
+  for (struct usb_bus *bus = usb_get_busses(); bus && !devh; bus = bus->next) {
+    for (struct usb_device *dev = bus->devices; dev; dev = dev->next) {
+      if (dev->descriptor.idVendor == vendor_id) {
+        devh = usb_open(dev);
+        if (!devh) {
+          fprintf(stderr, "Failed to open device with Product ID: %#04x\n", dev->descriptor.idProduct);
+          return CONTROL_ERROR;
+        }
+        current_pid = dev->descriptor.idProduct;
+  #else
```

Comment:
This is just too hard to read. Better to split it into architecture specific functions

Reply:
Just two separate code segments that are conditionally included would help readability.